app/models/manageiq/providers/kubevirt/inventory/collector.rb

```diff
@@ -43,11 +43,26 @@ def initialize_for_targeted_refresh
   end
 
   def initialize_for_full_refresh
-    @manager.with_provider_connection do |connection|
-      @nodes = connection.nodes
-      @vms = connection.vms
-      @vm_instances = connection.vm_instances
-      @templates = connection.templates
-    end
+    @nodes        = kube_client.get_nodes
+    @vms          = kubevirt_client.get_virtual_machines
+    @vm_instances = kubevirt_client.get_virtual_machine_instances
+    @templates    = openshift_template_client.get_templates
```

Comment:
More out of curiosity, and not for this PR, but could we collect these from the clients in parallel?